File: PA/Examples/06_splatka.cs
using System;

namespace ConsoleApp11
{
    class Program
    { 
        static void Main(string[] args)
        {
            int pocetLetSplaceni = 20;
            double rocniUrokProcenta = 2;
            double D = 1000000; // dluh

            Console.WriteLine($"Pocet let {pocetLetSplaceni}, urok {rocniUrokProcenta}% rocne a castka {D} Kc");
            
            // Vypocet splatky uroku dle vzorce http://www.aristoteles.cz/matematika/financni_matematika/hypoteka-vypocet.php
            
            int n = pocetLetSplaceni * 12; // pocet mesicu splaceni
            double i = rocniUrokProcenta / (12 * 100); // desetinne cislo

            double v = 1 / (1 + i);
            double splatka = (i * D) / (1 - Math.Pow(v, n));

            Console.WriteLine($"Mesicni splatka bude {splatka:F2} Kc");
        }
    }
}

File: Pripravka/2025/src/06_splatka.c
#include <stdio.h>
#define _USE_MATH_DEFINES
#include <math.h>

int main()
{
	double x = pow(2.0, 8.0); // 2^8 = 256.0

	double D = 6100000;
	double urokProcenta = 4.0;
	double i = urokProcenta / 100 / 12.0;
	int n = 30 * 12;

	double v = 1 / (1 + i);
	double A = i * D / (1 - pow(v, n));

	printf("Mesicni splatka: %lf\n", A);
}
